dask_cuda/utils.py

```diff
@@ -649,3 +653,116 @@ def get_gpu_uuid_from_index(device_index=0):
+        "memory_spill_fraction",
+        "memory_target_fraction",
+    ]:
+        ret[mem] = getattr(dask_worker.memory_manager, mem)
+
+    # jit unspilling set
+    ret["jit-unspill"] = isinstance(dask_worker.data, ProxifyHostFile)
+
+    # get optional device-memory-limit
+    if ret["jit-unspill"]:
+        ret["device-memory-limit"] = dask_worker.data.manager._device_memory_limit
+    else:
+        has_device = hasattr(dask_worker.data, "device_buffer")
+        if has_device:
+            ret["device-memory-limit"] = dask_worker.data.device_buffer.n
+
+    # using ucx ?
+    scheme, loc = parse_address(dask_worker.scheduler.address)
+    ret["protocol"] = scheme
+    if scheme == "ucx":
+        import ucp
+
+        ret["ucx-transports"] = ucp.get_active_transports()
+
+    # comm timeouts
+    ret["distributed.comm.timeouts"] = dask.config.get("distributed.comm.timeouts")
+    # ret.update(plugin_config)
+
+    return ret
+
+
+async def _get_sched_config(client):
+    worker_ttl = await client.run_on_scheduler(
+        lambda dask_scheduler: dask_scheduler.worker_ttl
+    )
+    extensions = list(
+        await client.run_on_scheduler(
+            lambda dask_scheduler: dask_scheduler.extensions.keys()
+        )
+    )
+    ret = {}
+    ret["distributed.scheduler.worker-ttl"] = worker_ttl
+    ret["active-extensions"] = extensions
+
+    return ret
+
+
+async def _get_cluster_configuration(client):
+    worker_config = await client.run(get_worker_config)
+    ret = await _get_sched_config(client)
+
+    # does the cluster have any workers ?
+    if worker_config:
+        w = list(worker_config.values())[0]
+        ret.update(w)
+
+    return ret
+
+
+def print_cluster_config(client):
+    """print current Dask cluster configuration"""
+    data = get_cluster_configuration(client)
+    formatted_byte_keys = [
+        "memory_limit",
+        "device-memory-limit",
+        "initial_pool_size",
+        "maximum_pool_size",
+    ]
+
+    try:
+        from rich.console import Console
+        from rich.table import Table
+    except ModuleNotFoundError as e:
+        error_msg = (
+            "Please install rich `python -m pip install rich` "
+            "to print a table of the current Dask Cluster Configuration"
+        )
+        raise ModuleNotFoundError(error_msg) from e
+
+    table = Table(title="Dask Cluster Configuration")
+    table.add_column("Parameter", justify="left", style="bold bright_green")
+    table.add_column("Value", justify="left", style="bold bright_green")
+
+    for key, value in sorted(data.items(), key=operator.itemgetter(0)):
+        if any([k in key for k in formatted_byte_keys]) and value is not None:
+            value = format_bytes(value)
+        table.add_row(key, str(value))
+    Console().print(table)
+
+
+def get_cluster_configuration(client):
+    data = client.sync(
+        _get_cluster_configuration, client=client, asynchronous=client.asynchronous
+    )
+    return data
```

Comment:
Actually, I think we need to go the other way and build an async print function.  This would allow of for usage of `print_cluster_config` inside a jupyter notebook with an async client